test/contract/auditing/auditing_test.go

```diff
@@ -0,0 +1,144 @@
+package auditing
+
+import (
+	"context"
+	_ "embed"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/auditing"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/control"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/launcher"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/contract"
+)
+
+//go:embed test.yml
+var testYml string
+
+const (
+	testVersion = "2.1.0"
+)
+
+func TestMain(m *testing.M) {
+	if !control.Enabled("AKO_CONTRACT_TEST") {
+		log.Printf("Skipping contract test as AKO_CONTRACT_TEST is unset")
+		return
+	}
+	l := launcher.NewFromEnv(testVersion)
+	l.Launch(
+		testYml,
+		launcher.WaitReady("atlasprojects/my-project", 30*time.Second))
```

Comment:
nit: 30 seconds might be quite aggressive

Reply:
coming next